ocaml/parsing/parser.mly

```diff
@@ -3499,9 +3499,18 @@ pattern_no_exn:
         { Ppat_alias($1, $3) }
     | self AS error
         { expecting $loc($3) "identifier" }
-    | pattern_comma_list(self) %prec below_COMMA
-        { Ppat_tuple(List.rev $1) }
-    | self COLONCOLON error
+  ) { $1 }
+  | reversed_labeled_tuple_pattern(self)
+      { let closed, pats = $1 in
+        if    closed = Closed
+           && List.for_all (fun (l,_) -> Option.is_none l) pats
+        then
+          mkpat ~loc:$sloc (Ppat_tuple(List.rev_map snd pats))
+        else
+          ppat_lttuple $sloc (List.rev pats) closed
+      }
+  | mkpat(
+      self COLONCOLON error
```

Comment:
Sorry I wasn't clear - I meant that currently there are two `mkpat` (but there was only one) - I think if we keep it that way, we can minimize the diff to only removing `pattern_comma_list` and adding `reversed_labeled_tuple_pattern`.

Reply:
Right - I think I see, and agree that is better.  Does it look right now?

Reply:
yeah I think it's perfect now - thanks for fullfilling my nits.